internal/coreinternal/attraction/type_converter.go

```diff
@@ -0,0 +1,57 @@
+package attraction
+
+import (
+	"strconv"
+
+	"go.opentelemetry.io/collector/model/pdata"
+)
+
+func convertValue(to string, v pdata.AttributeValue) {
+	switch to {
+	case "string":
+		switch v.Type().String() {
+		case "STRING":
```

Comment:
Instead of converting to String, it's better just to do switch on `v.Type()` and refer to the named constant value ("enum") type, e.g.
```suggestion
		case pdata.AttributeValueTypeString:
```